src/pages/players/[id]-[username]/statistics.astro

```diff
@@ -0,0 +1,132 @@
+---
+export const prerender = false
+import Layout from "../../../layouts/Layout.astro"
+import Header from "../../../components/layout/Header.astro"
+import HeaderContent from "../../../components/layout/HeaderContent.astro"
+import HeaderNav from "../../../components/layout/HeaderNav.astro"
+
+import { PlayersApi } from "../../../lib/api"
+import { RankedBadge } from "../../../components/ui/RankedBadge"
+import HeaderMeta from "../../../components/layout/HeaderMeta.astro"
+import Section from "../../../components/layout/Section.astro"
+import Main from "../../../components/layout/Main.astro"
+import { MmrHistoryChart } from "../../../components/widgets/MmrHistoryChart"
+import { Widget } from "../../../components/ui/Widget"
+import { FactionDropdown } from "../../../components/widgets/FactionDropdown"
+import { formatDateRelative } from "../../../lib/format"
+import { getDataOrErrorResponse } from "../../../lib/utils"
+import Sidebar from "../../../components/layout/Sidebar.astro"
+import PlayerActivity from "../../../components/widgets/PlayerActivity.astro"
+import PlayerMatchupStats from "../../../components/widgets/PlayerMatchupStats.astro"
+import PlayerGameLengthStats from "../../../components/widgets/PlayerGameLengthStats.astro"
+import PlayerOpponents from "../../../components/widgets/PlayerOpponents.astro"
+
+const playerId = Astro.params.id!
+const playerUsername = Astro.params.username!
+
+const [[player, playerMatches, playerActivity, playerMatchupStats, playerOpponents], error] =
+  await getDataOrErrorResponse(
+    PlayersApi.getPlayer({ playerId }),
+    PlayersApi.getPlayerMatches({ playerId, count: 1 }),
+    PlayersApi.getPlayerStatisticsActivity({ playerId }),
+    PlayersApi.getPlayerStatisticsMatchups({ playerId }),
+    PlayersApi.getPlayerStatisticsOpponents({ playerId, count: 10 })
+  )
+if (error) return error
+
+const leaderboardEntries = player?.leaderboard_entries
+  .filter((x) => x.matches > 0)
+  .sort((a, b) => (b.points ?? 0) - (a.points ?? 0))
+
+const faction = new URLSearchParams(Astro.url.search).get("faction") || leaderboardEntries?.[0].race
+
+const factionActivity = playerActivity.races.find((activity) => activity.aggregated?.race === faction)
+const league = leaderboardEntries.find((leaderboard) => leaderboard.race === faction) || leaderboardEntries?.[0]
+
+const mmrEntries: number[] = []
+const dateLabels: string[] = []
+if (factionActivity) {
+  for (const entry of factionActivity.history) {
+    if (entry.date && entry.mmr.median) {
+      dateLabels.push(entry.date)
+      mmrEntries.push(Math.round(entry.mmr.median))
+    }
+  }
+}
+const today = new Date().toISOString().slice(0, 10)
+if (dateLabels[dateLabels.length - 1] !== today) {
+  dateLabels.push(today)
+  mmrEntries.push(Math.round(league.mmr))
+} else {
+  mmrEntries[mmrEntries.length - 1] = Math.round(league.mmr)
+}
```

Comment:
I am doing that because:
- I want the graph to end today
- I want to last value to match the current MMR which is now not necessarily the case. I started talking to @reneklacan about it

Reply:
Second point is solved by the new endpoint 👍 
However, I am still thinking that adding today (if not present) might be a good idea to make sure the graph doesn't end at a random date. WDYT?